eureka2-core/src/main/java/com/netflix/eureka2/config/EurekaTransportConfig.java

```diff
@@ -0,0 +1,13 @@
+package com.netflix.eureka2.config;
+
+import com.netflix.eureka2.transport.EurekaTransports;
+
+/**
+ * @author David Liu
+ */
+public interface EurekaTransportConfig {
```

Comment:
We need a builder here, that plays well with EurekaBuilder.
For example:
```java
newEurekaBuilder().
    .withTransport(newTransportConfig().withCodec(Codec.Avro))
   .withXXX
```
EurekaBuilder should accept both EurekaTransportConfig and EurekaTransportConfigBuilder for client convenience.